main.go

```diff
@@ -100,6 +100,15 @@ func main() {
 	}
 	defer utils.Close(lf)
 
+	endpoint := os.Getenv("SPANNER_API_ENDPOINT")
+	emulatorHost := os.Getenv("SPANNER_EMULATOR_HOST")
+	if !((endpoint == "" && (emulatorHost == "" || emulatorHost == constants.SPANNER_EMULATOR_HOST)) || (endpoint == constants.STAGING_SPANNER_API_ENDPOINT && emulatorHost == "")) {
```

Comment:
This if condition is difficult to read. Suggest moving this to a 
func checkSpannerEndpoints() bool {
   endpoint := os.Getenv("SPANNER_API_ENDPOINT")
   emulatorHost := os.Getenv("SPANNER_EMULATOR_HOST")
  // list down the conditions separately.
}